build-tools-internal/src/main/groovy/elasticsearch.bwc-test.gradle

```diff
@@ -32,7 +32,7 @@ tasks.register("bwcTest") {
 }
 
 plugins.withType(ElasticsearchTestBasePlugin) {
-  tasks.withType(Test).configureEach {
+  tasks.withType(Test).matching { it.name ==~ /v[0-9\.]+#.*/ }.configureEach {
```

Comment:
In groovy, `==~` means match pattern `v[0-9\.]+#.*`, which should match any BWC test. These tests are in format: `v8.2.3#oneThirdUpgradedTest`. This is needed to avoid applying BWC plugin to any other non-bwc tests (e.g. javaRestTest)  if they are defined in the same project.

Reply:
Nice, thanks for explaining!